File: 49. Group Anagrams/main.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        ret_dict = defaultdict(list)
        for val in strs:
            ret_dict[tuple(sorted(val))].append(val)
        return list( ret_dict.values())
    
# Runtime: 188 ms
# Memory: 18 MB

File: 49. Group Anagrams/main.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string, vector<string>> hashmap;
        vector<vector<string>> ret_vec;
        for (string& str: strs){
            string sorted = str;
            sort(sorted.begin(), sorted.end());
            hashmap[sorted].push_back(str);
        }
        
        for (auto& kv: hashmap){
            ret_vec.push_back(move(kv.second));
        }
        return ret_vec;
    }
};

//Runtime: 32 ms
// Memory: 18.4 MB
